### DISCUSSION TITLE
Proposal: allow us to pass fetch option for working with SvelteKit

### DISCUSSION THREAD

**Discussion by dsm-72** [pocketbase/pocketbase#1609] [Q&A]
SvelteKit is giving me this warning

> Loading http://127.0.0.1:8090/api/collections/<test>/records/<id>?expand=<another> using window.fetch. For best results, use the fetch that is passed to your load function: https://kit.svelte.dev/docs/load#making-fetch-requests
where `127.0.0.1:8090` is my pocketbase server in development mode

where the request I am making is

```javascript
// src/lib/pocketbase.ts
export const pb = new PocketBase('http://127.0.0.1:8090');

// src/routes/test/[id]/+page.js
import {pb} from '$/lib/pocketbase'

/** @type {import('./$types').PageLoad} */
export async function load ({ params }) {
    const recordID:stirng = params.id;
    try {
        const record = await pb
            .collection('test')
            .getOne(recordID, {
                expand: 'another'
            })     
        console.log(record)
        // NOTE: returning record throws something about plaintext
        return {...record}
    } catch (err) {
        console.error(err)
        throw error(404, 'Not Found')
    } finally {
      // TODO
    }
}
```

so it would be nice if I could update `load` function like so

```javascript
export async function load ({ fetch, params }) {
    const recordID:stirng = params.id;
    try {
        const record = await pb
            .collection('test')
            .getOne(recordID, {
                expand: 'another'
            }, fetch)     
        // NOTE: returning record throws something about plaintext
        return {...record}
    }  // ...
}
```

**Comment by ganigeorgiev** [comment]
We can provide an option in the [JS SDK](https://github.com/pocketbase/js-sdk) but I'm not sure if it is worth doing it.

It seems that the only eventual benefit would be the last bullet point listed in https://kit.svelte.dev/docs/load#making-fetch-requests:

> during server-side rendering, the response will be captured and inlined into the rendered HTML. Note that headers will not be serialized, unless explicitly included via [filterSerializedResponseHeaders](https://kit.svelte.dev/docs/hooks#server-hooks-handle). Then, during hydration, the response will be read from the HTML, guaranteeing consistency and preventing an additional network request - if you got a warning in your browser console when using the browser fetch instead of the load fetch, this is why.

I'll think a little more on it but for now remains a very low priority.

**Reply by adamshand** [reply]
Does anyone know if there's a way to silence the warning message in the browser?

**Reply by ganigeorgiev** [reply]
This is an old discussion. Note that with JS SDK v0.17+ you can specify custom `fetch` request options - https://github.com/pocketbase/js-sdk?tab=readme-ov-file#custom-request-options.
